api/test/application/user/reset-password-post.js

```diff
@@ -0,0 +1,107 @@
+const Lab = require('@hapi/lab');
+const { expect } = require('@hapi/code');
+const { stringify } = require('querystring');
+const sinon = require('sinon');
+const uuid = require('uuid');
+
+const getServer = require('../../server');
+const { truncateAll } = require('../../helpers/db');
+const factory = require('../../helpers/factory');
+
+const { describe, it, before, beforeEach, after, afterEach } = (exports.lab =
+  Lab.script());
+
+describe('POST /user/reset-password', () => {
+  let server, client;
+
+  before(async () => {
+    server = await getServer();
+  });
+
+  beforeEach(async () => {
+    client = await factory.create('client');
+    await client.refresh({ withRelated: 'redirect_uris' });
+  });
+
+  after(async () => {
+    await truncateAll();
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('returns 403 forbidden if token is not supplied', async () => {
+    const res = await server.inject({
+      method: 'POST',
+      url: '/user/reset-password',
+    });
+
+    expect(res.statusCode).to.equal(403);
+  });
+
+  it('returns 400 when the payload is missing', async () => {
+    const user = await factory.create('user');
+    const token = await factory.create('emailToken', { user_id: user.id });
+
+    const query = {
+      token: token.get('token'),
+      client_id: client.get('client_id'),
+      redirect_uri: client.related('redirect_uris').at(0).get('uri'),
+    };
+
+    const res = await server.inject({
+      method: 'POST',
+      url: `/user/reset-password?${stringify(query)}`,
+    });
+
+    expect(res.statusCode).to.equal(400);
+  });
+
+  it('returns 400 when passwords differ', async () => {
+    const user = await factory.create('user');
+    const token = await factory.create('emailToken', { user_id: user.id });
+
+    const query = {
+      token: token.get('token'),
+      client_id: client.get('client_id'),
+      redirect_uri: client.related('redirect_uris').at(0).get('uri'),
+    };
+
+    const res = await server.inject({
+      method: 'POST',
+      url: `/user/reset-password?${stringify(query)}`,
+      payload: {
+        password: 'abcdefgh',
+        pass2: 'ijklmnop',
+      },
+    });
+
+    expect(res.statusCode).to.equal(400);
```

Comment:
This is where I think you could use the dom-testing-library to look for validation messages.

You can also get a little creative with this stuff and test behind the route as well. It might not be super straight forward, so maybe this is still best, but calling the controller or service functions or whatever might offer you some interesting fidelity into the things you want to test.